examples/exex/kona/src/main.rs

```diff
@@ -0,0 +1,244 @@
+                    "Validated Payload Attributes {derived_attributes_count} [L2 Block Num: {}] [L2 Timestamp: {}] [L1 Origin Block Num: {}]",
+                    attributes.parent.block_info.number + 1,
+                    attributes.attributes.timestamp,
+                    pipeline.origin().unwrap().number,
+                );
+                info!(target: "loop", "attributes: {:#?}", attributes);
+            } else {
+                debug!(target: "loop", "No attributes to validate");
+            }
+        } else if let Ok(s) = now.elapsed().map(|s| s.as_secs()) {
+            if s > 60 {
+                now = std::time::SystemTime::now();
+                let dist = if curr_tip <= cfg.genesis.l1.number {
+                    cfg.genesis.l1.number - curr_tip
+                } else {
+                    0
+                };
+                info!(target: "loop", "Not synced to l2 genesis. Distance to genesis: {}", dist);
+            }
+        }
+
+        if let Ok(notification) = ctx.notifications.try_recv() {
+            if let Some(reverted_chain) = notification.reverted_chain() {
+                chain_provider.commit(reverted_chain.clone());
+                let block_info = info_from_header(&reverted_chain.tip().block);
+                let blocks = reverted_chain
+                    .blocks_iter()
+                    .collect::<Vec<&reth_primitives::SealedBlockWithSenders>>();
+                for block in blocks {
+                    let tx_hashes = block
+                        .transactions()
+                        .map(|tx| tx.hash)
+                        .collect::<Vec<reth_primitives::TxHash>>();
+                    let blobs = ctx.pool().get_all_blobs(tx_hashes)?;
+                    let blobs = blobs
+                        .into_iter()
+                        .map(|b| b.1)
+                        .collect::<Vec<reth_primitives::BlobTransactionSidecar>>();
+                    let mut locked_blob_provider = blob_store
+                        .lock()
+                        .map_err(|_| eyre::eyre!("Failed to lock blob provider"))?;
+                    let blobs = blobs.into_iter().flat_map(|b| b.blobs).collect();
+                    locked_blob_provider.insert_blobs(block.hash(), blobs);
+                }
+                cursor = if let Some(c) = cursor_map.get(&block_info) {
+                    *c
+                } else {
+                    panic!("Failed to get previous cursor for old chain tip");
+                };
+                pipeline.reset(block_info).await.map_err(|e| eyre::eyre!(e))?;
+            }
+            if let Some(committed_chain) = notification.committed_chain() {
+                chain_provider.commit(committed_chain.clone());
+                let tip_number = committed_chain
+                    .blocks_iter()
+                    .map(|b| b.block.header.header().number)
+                    .max()
+                    .unwrap_or_default();
```

Comment:
```suggestion
                let tip_number = committed_chain.tip();
```

it will panic on an empty chain (should be impossible in the real world), and also I think `unwrap_or_default()` is incorrect here